File: Exercise_28.py

# Calculate the sum of the first 100 prime numbers
from Exercise_16 import is_prime
def sum_of_first_n_prime_numbers(n):
    sum = 0
    i = 1
    counter = 0
    array = []
    while counter < n:
        if is_prime(i):
            array.append(i)
            sum += i
            counter += 1
        i += 1
    return sum, array


print(f"Sum of first 100 prime numbers is {sum_of_first_n_prime_numbers(100)}")

File: Randy-JRA/Exercise_28.cpp
//Challenge 28 is to Calculate the sum of first 100 prime numbers

#include<iostream> 
using namespace std; 
 
int main() 
{ 
	int n=100, isprime = 1, count, i, num=3,sum=2; 
 	 
	if(n>=1) 
	{ 
		for (count=2; count<=n ; ) 
		{ 
			for(i=2; i<=num-1; i++) 
			{ 
				if(num%i==0) 
				{ 
					isprime = 0; 
					break; 
				} 
			} 
	 
			if(isprime) 
			{ 
				sum += num; 
				count++; 
			} 
	 
		isprime = 1; 
		num++;   	 
		} 
	} 
	 
	cout<<"The sum of first "<<n<<" prime numbers is "<<sum;		 
} 
